plugins/services/src/js/reducers/serviceForm/JSONReducers/Constraints.js

```diff
@@ -19,12 +21,18 @@ module.exports = {
     const constraints = findNestedPropertyInObject(state, "constraints") || [];
 
     if (!Array.isArray(constraints)) {
-      return [];
+      return [new Transaction(["constraints"], "not-list", ERROR)];
     }
 
     return JSONParser(
       constraints.reduce(function(memo, constraint) {
+        if (constraint == null) {
+          return memo;
+        }
+
         if (!Array.isArray(constraint)) {
+          memo.push({ error: true });
```

Comment:
💡 I'd push an `Error` instead of an object with an error flag, as it would provide a more secure error condition (see https://github.com/dcos/dcos-ui/pull/2671/files#diff-5b04656bc39045c79cea0891f97950cbR77) and  allow us to pass a message.

Reply:
Makes a lot of sense yes.